Add dropdown menu to Header and disclaimer dialog

Jvr2022 opened the pull request:
Replaces the fixed disclaimer banner with a dialog shown on first visit, and refactors the Header to use a dropdown menu for user actions. Adds a custom dropdown menu component using @radix-ui/react-dropdown-menu and updates dependencies accordingly.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Modal disclaimer now appears on first visit, can be dismissed and stays hidden on return.
  * New account dropdown for authenticated users offering Profile, Nieuw thema, GitHub, and Uitloggen actions.

* **UI/UX**
  * Header navigation reorganized: avatar-driven menu replaces the previous direct button; login flow unchanged for unauthenticated users.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Jvr2022 committed:
Add dropdown menu to Header and disclaimer dialog

Replaces the fixed disclaimer banner with a dialog shown on first visit, and refactors the Header to use a dropdown menu for user actions. Adds a custom dropdown menu component using @radix-ui/react-dropdown-menu and updates dependencies accordingly.

dandanthedev commented on app/layout.tsx:
```diff
@@ -23,29 +24,43 @@ export default function RootLayout({
+    if (!nextOpen) {
+      document.cookie = "stm_seenDisclaimer=1; max-age=31536000; path=/";
+    }
+  };
```
@Jvr2022 ik zou localStorage gebruiken ipv cookies. lijkt me wat cleaner

dandanthedev replied:
ook is handleOpenChange niet echt nodig lijkt me, kan ook gwn inline op onopenchange

dandanthedev commented on components/Header.tsx:
```diff
@@ -23,26 +27,66 @@ export default function Header() {
+                  GitHub
+                </a>
+              </DropdownMenuItem>
+              <DropdownMenuItem
+                onSelect={(e) => {
+                  e.preventDefault();
+                  signOut();
+                }}
+                variant="destructive"
+                className="flex items-center gap-2"
+              >
+                <LogOut className="size-4" />
+                Uitloggen
+              </DropdownMenuItem>
```
criticial issue is wel wat overdreven coderabbit maar oke

Jvr2022 committed:
Switch disclaimer tracking from cookies to localStorage

Replaces usage of cookies with localStorage for tracking whether the disclaimer has been seen. Updates logic for opening and closing the disclaimer dialog accordingly.
